src/store/plugins/presenceReporter.js

```diff
@@ -0,0 +1,25 @@
+import { Events } from 'quasar'
+
+export default store => {
+  let timer = null
+  const setAway = () => store.dispatch('presence/toggle/away', true)
+  const resetTimer = () => {
+    clearTimeout(timer)
+    store.dispatch('presence/toggle/away', false)
+    timer = setTimeout(setAway, 30 * 1000) // 30 seconds
+  }
+
+  // listen on events
+  document.onload = resetTimer
+  document.onmousemove = resetTimer
+  document.onmousedown = resetTimer // touchscreen presses
+  document.ontouchstart = resetTimer
+  document.onclick = resetTimer // touchpad clicks
+  document.onscroll = resetTimer // scrolling with arrow keys
+  document.onkeypress = resetTimer
```

Comment:
Using `addEventListener` calls is more proper way to add events listeners, and doesn't suffer from conflicts if something else adds events listeners

Reply:
Good catch. I didn't think about it very much after copying it from Stackoverflow...